### PR TITLE
BugFix: Nested List Input Coercion not matching GraphQL spec 

### PR DESCRIPTION
**Pull request by abhinand-c**
This PR fixes the issue mentioned in https://github.com/graphql/graphql-js/issues/3858 to align input coercion of nested list with the [GraphQL spec](https://spec.graphql.org/October2021/#sel-HAHjBJLDLEBAAADFAb0le).

#### Changes (Similar to https://github.com/graphql-python/graphql-core/pull/194)
- Adds check to validate and raises error for the condition
- Updated tests to reflect the new logic, and validate error

### CODE REVIEW

**Comment by abhinand-c** [comment]
Hi @Cito @IvanGoncharov 
I hope everything's alright here. Is there anything more to do? And could you please review this PR?

**Comment by yaacovCR** [comment]
Hi, see https://github.com/graphql/graphql-js/issues/3858#issuecomment-1458786732

TLDR I think a helpful next step might be to bring the underlying issue to the attention of the WG at the next meeting.

**Comment by abhinand-c** [comment]
Closing this PR as spec was updated to clarify the isssue, and no change in implementation is required.  

**Ref:** 
- https://github.com/graphql/graphql-js/issues/3858#issuecomment-2495638447
- https://github.com/graphql/graphql-spec/pull/1057

**Comment by yaacovCR** [comment]
Thanks @abhinand-c for bringing this to the forefront!
